segment/segment.py

```diff
@@ -1,52 +1,56 @@
 import os
 import cv2
+import logging
 
+logging.basicConfig(level=logging.DEBUG)
 
-# Segment video into respective frames.
-def extract_frames(pathIn):
-    print("Extracting frames from video ...")
-    vidcap = cv2.VideoCapture(pathIn)
-    fps = vidcap.get(cv2.CAP_PROP_FPS)
-    success, frame = vidcap.read()
-    frames = []
-    while success:
-        frames.append(frame)
-        success, frame = vidcap.read()
-    vidcap.release()
-    print("Total frames extracted : %d" % len(frames))
-    return fps, frames
-
-
-def extract_fps(path):
-    vidcap = cv2.VideoCapture(path)
-    fps = vidcap.get(cv2.CAP_PROP_FPS)
-    return fps
-
-def scale_frames(frames, scale_x=4, scale_y=4):
-    print("Scaling images")
-    scaled_frames = []
-    for frame in frames:
-       size =  (frame.shape[1],frame.shape[0])
-       scaled_frame = cv2.resize(frame, (0,0), fx=scale_x, fy=scale_y, interpolation=cv2.INTER_CUBIC)
-       scaled_frames.append(scaled_frame)
-    return scaled_frames
-
-# Combine respective frames into a video
-def frames_to_video(frames, fps, pathOut="./out.mp4"):
-    print('Combining frames')
-    fourcc = cv2.VideoWriter_fourcc('D', 'I', 'V', 'X')
-    out = cv2.VideoWriter(pathOut, fourcc, fps, frames[0].shape[:2][::-1])
-    for frame in frames:
-       out.write(frame)
-    out.release()
-
-def scale_video(video_location, dest_path="./out.mp4", scale_x=2, scale_y=2):
-    fps, frames = extract_frames(video_location)  # Choose your PathOut yourself.
-    scaled_frames = scale_frames(frames, scale_x, scale_y)
-    frames_to_video(scaled_frames, fps, dest_path)
-
-
-if __name__ == '__main__':
-    video_location = 'to_segment.mp4'
-    scale_video(video_location, dest_path="./out.mp4", scale_x=2, scale_y=2)
+class VideoScaler(object):
+
+    def __init__(self, path_in, path_out):
+        if  os.path.exists(path_in):
+            self.__path_in = path_in
+        else:
+            raise IOError('File does not exist: %s' % path_in)
+        self.__path_out  = path_out
 
+    # Segment video into respective frames.
+    def extract_frames(self):
+        logging.info("Extracting frames from video ...")
+        vidcap = cv2.VideoCapture(self.__path_in)
+        fps = vidcap.get(cv2.CAP_PROP_FPS)
+        success, frame = vidcap.read()
+        frames = []
+        while success:
+            frames.append(frame)
+            success, frame = vidcap.read()
+        vidcap.release()
+        logging.debug("Total frames extracted : %d" % len(frames))
+        return fps, frames
+
+
+    def scale_frames(self, frames, scale_x=4, scale_y=4):
+        logging.info('Scaling images')
+        scaled_frames = []
+        for frame in frames:
+           size =  (frame.shape[1],frame.shape[0])
+           scaled_frame = cv2.resize(frame, (0,0), fx=scale_x, fy=scale_y, interpolation=cv2.INTER_CUBIC)
+           scaled_frames.append(scaled_frame)
+        return scaled_frames
+
+    # Combine respective frames into a video
```

Comment:
Since the function name itself is self descriptive of what it does, the comment here is pretty redundant. For example,

n = 1 + 1 # adds two number
this comment is useless here. 

n = 3.1416 * y # Multiply by value of PI
Comment is useful.

Even better

PI = 3.1416
n = PI * y

As far as possible, comments should be replaced by self descriptive variable names. If you need to comment the "what" of some bunch of code, it might indicate that portion of code needs to have it's own function and so on. @Kamalpaneru